ncs/encrypt_script.py

```diff
@@ -232,148 +181,82 @@ def generate_suit_encryption_info(self, iv, encrypted_cek, string_key_id, output
-    plain_text_digest.bin - The digest of the plaintext firmware.
-    plain_text_size.txt - The size of the plaintext firmware in bytes.
-    """,  # noqa: W291, E501
-        formatter_class=RawTextHelpFormatter,
-    )
-
-    create_subparsers(parser)
-
-    arguments = parser.parse_args()
-
-    encrypted_asset = None
-    encrypted_cek = None
-
-    encryptor = Encryptor(arguments.kw_alg)
-
-    if arguments.command == "encrypt-and-generate":
-        encryptor.init_kms_backend(arguments.kms_script, arguments.context)
-        digest_generator = DigestGenerator(arguments.hash_alg.value)
-        digest_generator.generate_digest_size_for_plain_text(arguments.firmware, arguments.output_dir)
-        encrypted_asset, encrypted_cek = encryptor.generate_kms_artifacts(
-            arguments.firmware, arguments.key_name, arguments.context
+        encryption_info = self.generate_suit_encryption_info(init_vector, encrypted_cek, key_id)
+        return encrypted_content, tag, encryption_info
+
+    def _kw_alg_convert(self, kw_alg):
+        if kw_alg == SuitKWAlgorithms.A256KW:
+            self.cose_kw_alg = SuitCoseEncryptAlgorithms.COSE_ALG_A256KW.value
+        else:
+            self.cose_kw_alg = SuitCoseEncryptAlgorithms.COSE_ALG_DIRECT.value
+
+    def encrypt_and_generate(
+        self,
+        firmware: bytes,
+        key_name: str,
+        key_id: int,
+        context: str,
+        hash_alg: SuitDigestAlgorithms,
+        kw_alg: SuitKWAlgorithms,
+        kms_script: Path,
+    ) -> tuple[bytes, bytes, bytes, bytes, int]:
+        """
+        Encrypt the payload and return the encryption artifacts.
+
+        :param firmware: The plaintext firmware.
+        :param key_name: The name of the key used by the KMS to identify the key.
+        :param key_id: The key ID used to identify the key on the device.
+        :param context: Any context information that should be passed to the KMS backend during initialization
+                        and encryption.
+        :param hash_alg: The algorithm used to create plaintext digest.
+        :param kw_alg: Key wrap algorithm used to wrap the CEK.
+        :param kms_script: Python script containing a SuitKMS class with an encrypt function - used to communicate
+                           with a KMS.
+
+        :return: The encrypted payload, tag, encryption info, digest, and plaintext length.
+        :rtype: tuple[bytes, bytes, bytes, bytes, int]
+        """
+        self._kw_alg_convert(kw_alg)
+        self.init_kms_backend(kms_script, context)
+
+        digest_generator = DigestGenerator(hash_alg.value)
+        digest, plaintext_len = digest_generator.generate_digest_size_for_plain_text(firmware)
+        encrypted_asset, encrypted_cek = self.generate_kms_artifacts(firmware, key_name, context)
+        encrypted_payload, tag, encryption_info = self.generate_encryption_info_and_encrypted_payload(
+            encrypted_asset, encrypted_cek, key_id
```

Comment:
nit: the type of `encrypted_asset` in the `generate` method is `bytes`, but in the `generate_encryption_info_and_encrypted_payload` method, it is expected to be of type `Path`. This mismatch can lead to runtime errors. Please ensure that the type of `encrypted_asset` is consistent across these methods.